torch/csrc/distributed/c10d/control_plane/WorkerServer.cpp

```diff
@@ -0,0 +1,101 @@
+#include <httplib.h>
+#include <filesystem>
+#include <mutex>
+#include <shared_mutex>
+#include <tuple>
+#include <unordered_map>
+
+#include <ATen/core/interned_strings.h>
+#include <caffe2/utils/threadpool/WorkersPool.h>
+#include <torch/csrc/distributed/c10d/control_plane/WorkerServer.hpp>
+#include <torch/csrc/distributed/c10d/logging.h>
+
+namespace c10d {
+namespace control_plane {
+
+namespace {
+class RequestImpl : public Request {
+ public:
+  RequestImpl(const httplib::Request& req) : req_(req) {}
+
+  const std::string& body() override {
+    return req_.body;
+  }
+
+ private:
+  const httplib::Request& req_;
+};
+
+class ResponseImpl : public Response {
+ public:
+  ResponseImpl(httplib::Response& res) : res_(res) {}
+
+  void setStatus(int status) override {
+    res_.status = status;
+  }
+
+  void setContent(std::string&& content, const std::string& content_type)
+      override {
+    res_.set_content(std::move(content), content_type);
+  }
+
+ private:
+  httplib::Response& res_;
+};
+} // namespace
+
+WorkerServer::WorkerServer(const std::string& socketFile) {
+  // using unix sockets
+  server_.set_address_family(AF_UNIX);
+
+  server_.Get("/", [](const httplib::Request& req, httplib::Response& res) {
```

Comment:
What if we return list of available handlers?

Reply:
added `GET /handler/` and a html index pointing to it for humans